stream-chat-android-client/src/main/java/io/getstream/chat/android/client/clientstate/ClientStateService.kt

```diff
@@ -0,0 +1,129 @@
+package io.getstream.chat.android.client.clientstate
+
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.client.utils.Event
+import io.getstream.chat.android.client.utils.FiniteStateMachine
+
+internal class ClientStateService {
+    fun onConnected(user: User, connectionId: String) {
+        stateMachine.sendEvent(ChatClientEvent.ConnectedEvent(user, connectionId))
+    }
+
+    fun onDisconnected() {
+        stateMachine.sendEvent(ChatClientEvent.DisconnectedEvent)
+    }
+
+    fun onSetUser(user: User) {
+        stateMachine.sendEvent(ChatClientEvent.SetUserEvent(user))
+    }
+
+    fun onSetAnonymousUser() {
+        stateMachine.sendEvent(ChatClientEvent.SetAnonymousUserEvent)
+    }
+
+    fun onTokenReceived(token: String) {
+        stateMachine.sendEvent(ChatClientEvent.TokenReceivedEvent(token))
+    }
+
+    fun onDisconnectRequested() {
+        stateMachine.sendEvent(ChatClientEvent.DisconnectRequestedEvent)
+    }
+
+    private val stateMachine: FiniteStateMachine<ClientState, ChatClientEvent> by lazy {
+        FiniteStateMachine {
+            initialState(ClientState.Idle)
+
+            defaultHandler { state, event -> state.inappropriateStateError("handling event $event") }
```

Comment:
Looks like `inappropriateStateError` should just take `event` as its parameter.

Reply:
Good catch. Thanks